### ISSUE TITLE
Cart query always returns Tax Identifier value instead of Tax Title in the `applied_taxes` field.

### ISSUE THREAD

**Issue by homanp** [magento/magento2#30281]
### Preconditions (*)

1. 2.3.5-p1

### Steps to reproduce (*)

1. Add a product to cart.
2. Add shipping and billing address
3. Fetch cart via GraphQL query including `applied_taxes`.

### Expected result (*)

1. `applied_taxes` label should be that of the `Tax Title` instead of the `Tax Identifier` of the specific tax rate

### Actual result (*)

1. `applied_taxes` label always returns the value of the `Tax Identifier` instead of the localized `Tax Title`.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [x] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hello @homanp 

Thank you for your report.

We Confirm that the issue is reproducible on the latest 2.4-develop.

According to the Description of `applied_taxes` attribute, the query should return _"An array containing the **names** and amounts of taxes applied to the item"_ https://devdocs.magento.com/guides/v2.4/graphql/queries/cart.html#CartPrices

In the described case, the GrahQL query 
```graphql
{
  cart(cart_id: "7OwTsO6uJxQR71xm7tRwa7kHQ0ze16pt") {
    prices {
      applied_taxes {
        label
        amount{
          value
        }
      }
    }
  }
}
```
returns the name of the Tax Identifier, instead of the Tax Rule Name
![tax_ident](https://user-images.githubusercontent.com/51680745/107933577-940bc700-6f87-11eb-9d47-290a2778053d.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-40891` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @homanp. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-40891` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.

**Comment by homanp** [comment]
🙏  @magento-engcom-team

**Comment by hostep** [comment]
@homanp: could you confirm if this is really fixed in 2.4.3?

I can't find any reference to ticket number `MC-40891` here in github and on the [2.4.3 release notes](https://devdocs.magento.com/guides/v2.4/release-notes/open-source-2-4-3.html), nothing is mentioned about this bug.
